compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java

```diff
@@ -241,14 +255,24 @@ public void visit(BLangResource resourceNode) {
 
         validateReturnParameter(resourceNode, null, false);
         validateReferences(resourceNode);
+        validateDeprecationDocumentation(resourceNode.markdownDocumentationAttachment, false, resourceNode.pos, false);
```

Comment:
Why are we passing `false` for `isDeprecationAnnotationAvailable` and `isExpected`? 

A resource is a function (resource function) so the annotation is allowed.

Reply:
for resource, deprecation is not supported.

Reply:
The following builds for me on 1.2.0. Was it disallowed later?

```ballerina
import ballerina/http;

service serviceName on new http:Listener(8080) {

    @deprecated
    resource function newResource(http:Caller caller, http:Request request) {
        
    }
}
```

Reply:
It is still working. Yes, I misunderstood. That validation is not needed.